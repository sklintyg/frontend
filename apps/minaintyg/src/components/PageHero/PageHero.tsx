import { IDSIconSuccess } from '@frontend/ids-react-ts'
import { ReactNode } from 'react'

export function PageHero({ heading, type, children }: { heading: ReactNode; type: 'success' | 'error'; children: ReactNode }) {
  return (
    <div className="ids-content mx-auto flex max-w-[620px] flex-col gap-10 pt-12 text-center md:pt-24">
      {type === 'success' && (
        <IDSIconSuccess
          inline
          className="mx-auto"
          width="70px"
          height="70px"
          color="var(--color-grass-dark)"
          color2="var(--color-grass-base)"
        />
      )}
      <div>
        <h1 className="ids-heading-1 text-center">{heading}</h1>
        {children}
      </div>
    </div>
  )
}
