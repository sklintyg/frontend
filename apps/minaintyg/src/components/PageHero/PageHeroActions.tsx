import { ReactNode } from 'react'

export function PageHeroActions({ children }: { children: ReactNode }) {
  return (
    <div className="mb-5 text-left md:text-center">
      <div className="mb-5 mt-8 w-[43px] border-b border-stone-clear md:hidden" />
      <div className="mx-auto inline-flex flex-col gap-2.5 md:flex-row">{children}</div>
    </div>
  )
}
