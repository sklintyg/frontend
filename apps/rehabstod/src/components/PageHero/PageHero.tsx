import { IDSIconAttention, IDSIconUser } from '@inera/ids-react'
import type { ReactNode } from 'react'

export function PageHero({ type, children }: { type?: 'error' | 'user'; children?: ReactNode }) {
  return (
    <div className="px-5 md:p-0">
      <div className="ids-content mx-auto max-w-screen-md py-24 text-center">
        {type === 'error' && (
          <div className="mb-10 inline-block text-error-40">
            <IDSIconAttention height="4.375rem" width="4.375rem" color="currentColor" color2="currentColor" />
          </div>
        )}
        {type === 'user' && (
          <div style={{ background: '#c6d2df' }} className="mb-7 inline-block rounded-full px-6 py-5">
            <IDSIconUser height="3.75rem" width="3.125rem" color="#396291" color2="#3b4266" />
          </div>
        )}
        {children}
      </div>
    </div>
  )
}
