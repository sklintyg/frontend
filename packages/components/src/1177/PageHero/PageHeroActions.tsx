import { ReactNode } from 'react'
import { hasNoChildren } from '../../utils/hasNoChildren'

export function PageHeroActions({ children }: { children: ReactNode }) {
  if (hasNoChildren(children)) {
    return null
  }

  return (
    <div className="mb-5 text-left md:text-center">
      <div className="mb-5 mt-8 w-[43px] border-b border-page-hero-border-color md:hidden" />
      <div className="mx-auto inline-flex flex-col gap-2.5 md:flex-row">{children}</div>
    </div>
  )
}
