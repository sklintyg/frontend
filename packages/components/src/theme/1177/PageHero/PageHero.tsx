import type { ReactNode } from 'react'
import { Heading } from '../../../Heading'
import { Icon } from '../../../Icon'

export function PageHero({ heading, type, children }: { heading: ReactNode; type: 'success' | 'error' | 'none'; children: ReactNode }) {
  return (
    <div className="ids-content mx-auto flex max-w-[620px] flex-col gap-10 pt-12 text-center md:pt-24">
      <div
        className="mx-auto"
        style={{ fontSize: '72px', color: type === 'success' ? 'var(--IDS-COLOR-SUCCESS-40)' : 'var(--IDS-COLOR-PRIMARY-40)' }}
      >
        {type === 'success' && <Icon icon="success" />}
        {type === 'error' && <Icon icon="attention" />}
      </div>

      <div>
        <Heading level={1} size="xxl" className="text-center">
          {heading}
        </Heading>
        {children}
      </div>
    </div>
  )
}
