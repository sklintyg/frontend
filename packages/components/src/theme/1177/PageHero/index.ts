export * from './PageHero'
export * from './PageHeroActions'
