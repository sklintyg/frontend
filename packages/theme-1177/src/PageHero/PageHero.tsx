import { Heading, Icon } from '@frontend/components'
import type { ReactNode } from 'react'

export function PageHero({ heading, type, children }: { heading: ReactNode; type: 'success' | 'error' | 'none'; children: ReactNode }) {
  return (
    <div className="ids-content mx-auto flex max-w-[620px] flex-col gap-10 pt-12 text-center md:pt-24">
      <div
        className="mx-auto"
        style={{ fontSize: '72px', color: type === 'success' ? 'var(--IDS-COLOR-SUCCESS-40)' : 'var(--IDS-COLOR-PRIMARY-40)' }}
      >
        {type === 'success' && <Icon icon="success" />}
        {type === 'error' && <Icon icon="attention" />}
      </div>

      <div>
        <Heading level={1} size="xxl" className="text-center">
          {heading}
        </Heading>
        {children}
      </div>
    </div>
  )
}
