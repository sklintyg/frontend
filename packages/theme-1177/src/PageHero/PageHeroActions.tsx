import { hasNoChildren } from '@frontend/components'
import type { ReactNode } from 'react'

export function PageHeroActions({ children }: { children: ReactNode }) {
  if (hasNoChildren(children)) {
    return null
  }

  return (
    <div className="mb-5 text-left md:text-center">
      <div style={{ borderColor: 'var(--color-stone-clear)' }} className="mb-5 mt-8 w-[43px] border-b md:hidden" />
      <div className="mx-auto inline-flex flex-col gap-2.5 md:flex-row">{children}</div>
    </div>
  )
}
