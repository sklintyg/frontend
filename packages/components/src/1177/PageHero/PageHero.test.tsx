import { render } from '@testing-library/react'
import { expect, it } from 'vitest'
import { PageHero } from './PageHero'

it('Should render as expected with success icon', () => {
  const { container } = render(
    <PageHero heading="heading" type="success">
      content
    </PageHero>
  )
  expect(container).toMatchSnapshot()
})

it('Should render as expected with attention icon', () => {
  const { container } = render(
    <PageHero heading="heading" type="error">
      content
    </PageHero>
  )
  expect(container).toMatchSnapshot()
})
