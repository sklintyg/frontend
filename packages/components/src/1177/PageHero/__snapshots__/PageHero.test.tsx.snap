// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render as expected with attention icon 1`] = `
<div>
  <div
    class="ids-content mx-auto flex max-w-[620px] flex-col gap-10 pt-12 text-center md:pt-24"
  >
    <ids-icon-attention
      class="mx-auto ids-icon"
    />
    <div>
      <h1
        class="ids-heading-1 text-center"
      >
        heading
      </h1>
      content
    </div>
  </div>
</div>
`;

exports[`Should render as expected with success icon 1`] = `
<div>
  <div
    class="ids-content mx-auto flex max-w-[620px] flex-col gap-10 pt-12 text-center md:pt-24"
  >
    <ids-icon-success
      class="mx-auto ids-icon"
    />
    <div>
      <h1
        class="ids-heading-1 text-center"
      >
        heading
      </h1>
      content
    </div>
  </div>
</div>
`;
